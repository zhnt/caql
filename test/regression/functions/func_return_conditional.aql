function max(a, b) {
    if (a > b) {
        return a
    } else {
        return b
    }
}

function min(x, y) {
    if (x < y) {
        return x
    }
    return y
}

function abs(n) {
    if (n < 0) {
        return -n
    }
    return n
}

function sign(value) {
    if (value > 0) {
        return 1
    } elif (value < 0) {
        return -1
    } else {
        return 0
    }
}

print("max(15, 8) = " + string(max(15, 8)))
print("min(15, 8) = " + string(min(15, 8)))
print("abs(-42) = " + string(abs(-42)))
print("abs(42) = " + string(abs(42)))
print("sign(10) = " + string(sign(10)))
print("sign(-5) = " + string(sign(-5)))
print("sign(0) = " + string(sign(0)))
