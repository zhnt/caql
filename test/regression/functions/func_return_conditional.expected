max(15, 8) = 15
min(15, 8) = 8
abs(-42) = 42
abs(42) = 42
sign(10) = 1
sign(-5) = -1
sign(0) = 0


